src/status_im/contexts/wallet/wallet_connect/responding_events.cljs

```diff
@@ -54,12 +70,54 @@
              :address    address
              :data       raw-data
              :version    typed-data-version
-             :on-error   #(rf/dispatch [:wallet-connect/on-sign-error %])
-             :on-success #(rf/dispatch [:wallet-connect/send-response %])}]]})))
+             :on-error   #(rf/dispatch [:wallet-connect/on-sign-error :screen/wallet-connect.sign-message
+                                        %])
+             :on-success #(rf/dispatch [:wallet-connect/send-response :screen/wallet-connect.sign-message
+                                        %])}]]})))
+
+(rf/reg-event-fx
+ :wallet-connect/respond-build-transaction
+ (fn [{:keys [db]} [screen on-success]]
+   (let [{:keys [raw-data]} (get db :wallet-connect/current-request)
+         chain-id           (-> raw-data
+                                (get-in [:params :chainId])
```

Comment:
Can `chainId` ever be undefined ?

Reply:
Have no idea — not like web3wallet has a good documentation :D
But nothing will work without the `chainId` — you literally cannot sign the transaction and accept it in case it's undefined.